chips/nrf5x/src/rtc.rs

```diff
@@ -1,28 +1,102 @@
 //! RTC driver, nRF5X-family
 
-use core::cell::Cell;
 use core::mem;
+use kernel::common::cells::OptionalCell;
+use kernel::common::regs::{ReadOnly, ReadWrite, WriteOnly};
 use kernel::hil::time::{self, Alarm, Freq32KHz, Time};
 use kernel::hil::Controller;
-use peripheral_registers::{RTC1, RTC1_BASE};
 
-fn rtc1() -> &'static RTC1 {
+pub const RTC1_BASE: usize = 0x40011000;
+
+#[repr(C)]
+struct RtcRegisters {
+    /// Start RTC Counter.
+    tasks_start: WriteOnly<u32, Task::Register>,
+    /// Stop RTC Counter.
+    tasks_stop: WriteOnly<u32, Task::Register>,
+    /// Clear RTC Counter.
+    tasks_clear: WriteOnly<u32, Task::Register>,
+    /// Set COUNTER to 0xFFFFFFF0.
+    tasks_trigovrflw: WriteOnly<u32, Task::Register>,
+    _reserved0: [u8; 240],
+    /// Event on COUNTER increment.
+    events_tick: ReadWrite<u32, Event::Register>,
+    /// Event on COUNTER overflow.
+    events_ovrflw: ReadWrite<u32, Event::Register>,
+    _reserved1: [u8; 56],
+    /// Compare event on CC[n] match.
+    events_compare: [ReadWrite<u32, Event::Register>; 4],
+    _reserved2: [u8; 436],
+    /// Interrupt enable set register.
+    intenset: ReadWrite<u32, Inte::Register>,
+    /// Interrupt enable clear register.
+    intenclr: ReadWrite<u32, Inte::Register>,
+    _reserved3: [u8; 52],
+    /// Configures event enable routing to PPI for each RTC event.
+    evten: ReadWrite<u32, Inte::Register>,
+    /// Enable events routing to PPI. The reading of this register gives the value of EV
+    evtenset: ReadWrite<u32, Inte::Register>,
+    /// Disable events routing to PPI. The reading of this register gives the value of E
+    evtenclr: ReadWrite<u32, Inte::Register>,
+    _reserved4: [u8; 440],
+    /// Current COUNTER value.
+    counter: ReadOnly<u32>,
+    /// 12-bit prescaler for COUNTER frequency (32768/(PRESCALER+1)). Must be written wh
```

Comment:
Parse error I guess should be `12 bit prescaler for COUNTER frequency (32768/(PRESCALER+1)).Must be written when RTC is stopped`